core/src/main/scala/eu/shiftforward/apso/io/S3FileDescriptor.scala

```diff
@@ -111,10 +111,11 @@ case class S3FileDescriptor(
   }
 
   private lazy val isDirectoryRemote = bucket.isDirectory(builtPath)
+  private lazy val isBucketRoot = elements.isEmpty && bucket.isBucket
```

Comment:
This is more of a `isBucketAndExists` method (a given path `s3://some-bucket-name` is necessarily a pointer to a bucket but it may not exist, just as `file:///some-non-existing-dir` might be a pointer to a directory)
This method does not seem very useful on the class, we might just use its definition inside the `exists` method body.

What do you think?